protoc-gen-kotlin/lib/src/commonMain/kotlin/pbandk/gen/CodeGenerator.kt

```diff
@@ -6,14 +6,15 @@ import pbandk.wkt.FieldOptions
 open class CodeGenerator(
     val file: File,
     val kotlinTypeMappings: Map<String, String>,
-    @Suppress("unused") val params: Map<String, String>
+    val params: Map<String, String>
 ) {
     protected val bld = StringBuilder()
     protected var indent = ""
 
     fun generate(): String {
         line("@file:OptIn(pbandk.PublicForGeneratedCode::class)").line()
         file.kotlinPackageName?.let { line("package $it") }
+        if (params["js_export"] == "true") line().line("import kotlin.js.JsExport")
```

Comment:
I am in no way an expert in Kotlin/JS so take my words below with a grain of salt.

I don't think you need `@JsExport` to work with `pbandk` from Kotlin/JS with the new IR backend, I believe it works as is. My personal use-case is making use of [the generation of typescript declarations](https://kotlinlang.org/docs/js-ir-compiler.html#preview-generation-of-typescript-declaration-files-d-ts) and stopping Kotlin/JS from mangling JavaScript names for classes and functions which is very helpful when you use a Kotlin/JS library from JavaScript.

Anyway, I do not actually have a strong opinion on this and, if you think that `@JsExport` should be enabled by default, I am more than happy to do so.

Reply:
Thanks for the explanation. That helps.

While we're not using pbandk's Kotlin/JS runtime internally at the moment, we are big users of Typescript. Moreso, I think any JavaScript support in pbandk should aim for maximum Typescript support out of the box as well. So let's just include the `@JsExport` annotations by default. We can add a param in the future to disable it if anyone reports a use case where `@JsExport` causes problems.

I wonder if it'd be sufficient to add a `@file:JsExport` annotation at the top instead of annotating each class individually?

Reply:
So I have tried making pbandk annotate generated classes with `@JsExport` by default, but unfortunately there is an  issue with `CodeGeneratorTest`. The generated code cannot be compiled by [kotlin-compile-testing](https://github.com/tschuchortdev/kotlin-compile-testing) as it compiles for JVM be default and `@JsExport` cannot be resolved there. I could not find a way to configure it so that it compiles as if the given source was a **common** source, maybe you have more experience with kotlin-compile-testing?

Reply:
According to the docs, it looks like `@JsExport` is only available in common and JS source sets, but not JVM or Native. So I guess we can't use it in generated pbandk code since code generated by pbandk needs to be useable not only in a common source set but also directly in any platform-specific source set (such as JVM).

The only idea I can come up with is to define our own annotation that is available for all platforms and aliased to `@JsExport` on Kotlin/JS. I.e. add a new `expect` annotation to pbandk's `commonMain` source set:

```kt
package pbandk

@Target([AnnotationTarget.CLASS, AnnotationTarget.PROPERTY, AnnotationTarget.FUNCTION, AnnotationTarget.FILE])
expect annotation class Export()
```

Then in `jsMain`, add an alias:

```kt
package pbandk

actual typealias Export = kotlin.js.JsExport
```

And in the `commonJvmAndroid` and `nativeMain` source sets define our own annotation that mimics the `@JsExport` one:

```kt
package pbandk

@Target([AnnotationTarget.CLASS, AnnotationTarget.PROPERTY, AnnotationTarget.FUNCTION, AnnotationTarget.FILE])
actual annotation class Export()
```

Then in the generated code pbandk would use the `@pbandk.Export` annotation instead of the `@JsExport` annotation. The syntax in my examples above might not be quite right, but hopefully it gets the idea across.

Reply:
Thanks, that is a neat trick!

Reply:
Oh, by the way, I had to change the API since we are introducing a new annotation, is that OK?